static/js/upload.js

```diff
@@ -92,19 +97,47 @@ export function get_item(key, config) {
 
 export function hide_upload_status(config) {
     get_item("send_button", config).prop("disabled", false);
-    get_item("send_status", config).removeClass("alert-info").hide();
+    get_item("send_status", config).remove();
+}
+
+function show_upload_banner(config, banner_type, banner_text) {
+    // We only show one upload banner at a time per compose box,
+    // and all uploads are combined into the same progress bar.
+    // TODO: It would be nice to separate the error banner into
+    // a different element, so that we can show it at the same
+    // time as the upload bar and other uploads can still continue
+    // when an error occurs.
+    const $upload_banner = get_item("send_status", config);
+    if ($upload_banner.length) {
+        if (banner_type === "error") {
+            // Hide moving bar so that it doesn't do the 1s transition to 0
```

Comment:
I don't super love this but couldn't think of a better thing to do.

Well, I guess I can take time in this PR to separate the error and upload banners out :p I'd rather do that in a followup though

Reply:
This seems fine -- we also need to fix the handling of multiple simultaneous uploads, and that'll likely be a related refactor.